src/Symfony/Component/Notifier/Bridge/GoogleChat/Component/ComponentTrait.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\GoogleChat\Component;
+
+use Symfony\Component\Notifier\Bridge\GoogleChat\Exception\NotEndedComponentException;
+
+/**
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ *
+ * @internal
+ */
+trait ComponentTrait
+{
+    protected $options = [];
+
+    /**
+     * @var ComponentInterface|null
+     */
+    protected $dirty;
```

Comment:
I used a very fluent syntax for building rich cards, inspired by how bundle configuration are defined. 

In order to improve DX, I flag components as "durty" when a child component is created but not ended yet. Throwing an exception if we try to create a new child component or end its parent component.
Tests contains examples of such errors. https://github.com/symfony/symfony/pull/36488/files#diff-6a50cd9b91e0692df17324efc963031bR173

Reply:
Removed. To be proposed in an other PR.